### PR TITLE
gh-588 Fix issues

### PR DESCRIPTION
**Pull request by MouazAlzahabi**

### CODE REVIEW

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift [hunk 1]
```diff
@@ -60,11 +66,17 @@ struct SelectOwnerAddressView: View {
                                             derivePrivateKey: true)?.privateKey else { return }
         do {
             try App.shared.keychainService.save(data: pkData, forKey: KeychainKey.ownerPrivateKey.rawValue)
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift:68]: I think having this code inside the view is not the right place, this should be either in a model or view model.

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift [hunk 2]
```diff
@@ -10,22 +10,26 @@ import SwiftUI
...
+    init(rootNode: HDNode?, onSubmit: (() -> Void)? = nil ) {
         self.rootNode = rootNode
-        _rootIsActive = rootIsActive
-        let addresses: [Address] = (0..<20).compactMap { self.addressAt($0) }
+        let addresses: [Address] = generateAddressesPage()
         _addresses = State(initialValue: addresses)
+        self.onSubmit = onSubmit
     }
 
     private func addressAt(_ index: Int) -> Address? {
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift]: - this should be in a view model
- the errors here are not handled - will crash if the private key from the data fails (line 39)

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift [hunk 3]
```diff
@@ -60,11 +66,17 @@ struct SelectOwnerAddressView: View {
                                             derivePrivateKey: true)?.privateKey else { return }
         do {
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift]: this logic should be in a view model

**Review by DmitryBespalov — CHANGES_REQUESTED** [review]

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift (hunk 2 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift]: @DmitryBespalov There is no view model defined for this screen at all, should we define it in this ticket ?

Comment (DmitryBespalov) [reply Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift]: could be in the next pr 👍

**Review by DmitryBespalov — APPROVED** [review]
Please make a subsequent PR with moving the logic into view model

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    private func addressAt(_ index: Int) -> Address? {
+        guard let pkData = rootNode?.derive(index: UInt32(index), derivePrivateKey: true)?.privateKey else {
+            return nil
+        }
+
+        do {
+            let address = try EthereumPrivateKey(hexPrivateKey: pkData.toHexString()).address
+            return Address(address, index: index)
+        } catch {
+            return nil
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift:42]: ok, this is just dismissing the error - maybe we should post a snackbar message?

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+    
+    func generateAddressesPage() {
+        addresses += (1...pageSize).compactMap { addressAt($0 + addresses.count) }
+    }
+
+    func importWallet() -> Bool {
+        guard let pkData = rootNode?.derive(index: UInt32(selectedIndex),
+                                            derivePrivateKey: true)?.privateKey else { return false }
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift]: the private key derivation is repeated, could be extracted into a separate helper function

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    private func addressAt(_ index: Int) -> Address? {
+        guard let pkData = rootNode?.derive(index: UInt32(index), derivePrivateKey: true)?.privateKey else {
+            return nil
+        }
+
+        do {
+            let address = try EthereumPrivateKey(hexPrivateKey: pkData.toHexString()).address
+            return Address(address, index: index)
```
Comment (sche) [inline Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift:39]: we decided to show indexes starting with 1; so you can use here `return Address(address, index: index + 1)`

File: Multisig/UI/Settings/App Settings/BasicAppSettingsView.swift [hunk 7]
```diff
@@ -82,11 +82,10 @@ struct BasicAppSettingsView: View {
                 }
             }
         } else {
-            NavigationLink(destination: EnterSeedPhraseView(rootIsActive: self.$addOwnerIsActive),
+            NavigationLink(destination: EnterSeedPhraseView(),
                            isActive: self.$addOwnerIsActive) {
```
Comment (sche) [inline Multisig/UI/Settings/App Settings/BasicAppSettingsView.swift]: do we still need addOwnerIsActive?

**Review by sche — APPROVED** [review]
nice

File: Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift (hunk 6 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift:39]: I have just made update, index will start from 0 but in UI will be displayed starting from 1

File: Multisig/UI/Settings/App Settings/BasicAppSettingsView.swift (hunk 7 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/App Settings/BasicAppSettingsView.swift]: Just tried it, and yes we don't need it anymore

**Pull request merged**